Mathlib/GroupTheory/GroupAction/MultipleTransitivity.lean

```diff
@@ -0,0 +1,313 @@
+    Function.Injective (hf.mulActionHom_embedding (ι := ι)) := by
+  intro _ _ hxy
+  ext
+  apply hf
+  simp only [← hf.mulActionHom_embedding_apply, hxy]
+
+variable (hf' : Function.Bijective f)
+
+@[to_additive]
+theorem Function.Bijective.mulActionHom_embedding_isBijective (hf : Function.Bijective f) :
+    Function.Bijective (hf.injective.mulActionHom_embedding (ι := ι)) := by
+  refine ⟨hf.injective.mulActionHom_embedding_isInjective, ?_⟩
+  intro y
+  obtain ⟨g, _, hfg⟩ := Function.bijective_iff_has_inverse.mp hf
+  use ⟨g ∘ y, hfg.injective.comp (EmbeddingLike.injective y)⟩
+  ext
+  simp only [hf.injective.mulActionHom_embedding_apply, coeFn_mk, comp_apply]
+  exact hfg (y _)
+
+end Functoriality
+
+namespace MulAction
+
+open scoped BigOperators Pointwise Cardinal
+
+variable (G α : Type*) [Group G] [MulAction G α]
+
+/-- An action of a group on a type α is n-pretransitive
+if the associated action on (Fin n ↪ α) is pretransitive. -/
+@[to_additive "An additive action of an additive group on a type α
+is n-pretransitive if the associated action on (Fin n ↪ α) is pretransitive."]
+abbrev IsMultiplyPretransitive (n : ℕ) := IsPretransitive G (Fin n ↪ α)
+
+variable {G H α β : Type*} [Group G] [MulAction G α] [Group H] [MulAction H β]
+  {σ : G → H} {f : α →ₑ[σ] β} (hf : Function.Injective f)
+
+/- If there exists a surjective equivariant map `α →ₑ[σ] β`
+then pretransitivity descends from `n ↪ α` to `n ↪ β`.
+
+The subtlety is that if it is not injective, this map does not induce
+an equivariant map from `n ↪ α` to `n ↪ β`. -/
+@[to_additive]
+theorem IsPretransitive.of_embedding {n : Type*}
+    (hf : Function.Surjective f) [IsPretransitive G (n ↪ α)] :
+    IsPretransitive H (n ↪ β) where
+  exists_smul_eq x y := by
+    let aux (x : n ↪ β) : (n ↪ α) :=
+      x.trans (Function.Embedding.ofSurjective (⇑f) hf)
+    have aux_apply (x : n ↪ β) (i : n) : f.toFun (aux x i) = x i := by
+      simp only [trans_apply, aux]
+      apply Function.surjInv_eq
+    obtain ⟨g, hg⟩ := exists_smul_eq (M := G) (aux x) (aux y)
+    use σ g
+    ext i
+    rw [DFunLike.ext_iff] at hg
+    rw [smul_apply]
+    simp only [← aux_apply, ← hg, smul_apply, MulActionHom.map_smul']
+
+@[to_additive]
+theorem IsPretransitive.of_embedding_congr {n : Type*}
+    (hσ : Function.Surjective σ) (hf : Function.Bijective f) :
+    IsPretransitive G (n ↪ α) ↔ IsPretransitive H (n ↪ β) :=
+  isPretransitive_congr hσ hf.mulActionHom_embedding_isBijective
+
+section Zero
+
+/-- Any action is 0-pretransitive. -/
+@[to_additive]
+theorem is_zero_pretransitive {n : Type*} [IsEmpty n] :
+    IsPretransitive G (n ↪ α) := inferInstance
+
+/-- Any action is 0-pretransitive. -/
+@[to_additive]
+theorem is_zero_pretransitive' :
+    IsMultiplyPretransitive G α 0 := inferInstance
+
+
+end Zero
+
+section One
+
+/-- The equivalence `one ↪ α` with `α`, for `Unique one`. -/
+def _root_.Function.Embedding.oneEmbeddingEquiv {one : Type*} [Unique one] : (one ↪ α) ≃ α where
```

Comment:
This (and probably the results below) can go to `Mathlib.Logic.Embedding.Basic`.

Reply:
The first went there. For the two other, I still need to find a place for them.

Reply:
Finally, I left them here. The point is that the initial files do not yet allow equivariant maps.

Reply:
OK, no problem.